cpp/arcticdb/version/version_core.cpp

```diff
@@ -277,6 +277,17 @@ VersionedItem delete_range_impl(
     return versioned_item;
 }
 
+void sorted_data_check(InputTensorFrame& frame, index::IndexSegmentReader& index_segment_reader){
+    bool input_data_is_sorted = frame.desc.get_sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING || 
+                                frame.desc.get_sorted() == arcticdb::proto::descriptors::SortedValue::UNKNOWN;
+    bool existing_data_is_sorted = index_segment_reader.mutable_tsd().stream_descriptor().sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING || 
+                                    index_segment_reader.mutable_tsd().stream_descriptor().sorted() == arcticdb::proto::descriptors::SortedValue::UNKNOWN;
+    bool is_time_series = std::holds_alternative<stream::TimeseriesIndex>(frame.index);
+    sorting::check<ErrorCode::E_UNSORTED_DATA>(
+        input_data_is_sorted && existing_data_is_sorted && is_time_series,
+        "When calling update on sorted data, the input data must be sorted.");
```

Comment:
As @jamesmunro has suggested, let's split this into three seperate checks.

Reply:
done!